src/lib/sql/sql_pipeline.cpp

```diff
@@ -53,25 +55,19 @@ SQLPipeline::SQLPipeline(const std::string& sql, std::shared_ptr<TransactionCont
       }
     }
 
-    try {
-      auto parsed_statement = std::make_shared<hsql::SQLParserResult>(statement);
-
-      // This statements is now owned by the parse result, which will free the memory.
-      num_statements_released++;
+    const auto statement_string_length = statement->stringLength;
```

Comment:
Try to keep the declaration as close as possible to the first use of the variable. The next two lines deal with a different thing.

Reply:
Moved it, ad now will move it back here. statement is null after releasing it from the unique pointer, so we need to access it here.